common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -772,11 +774,16 @@ def _load_items(self, course_entry, block_keys, depth=0, **kwargs):
         Load the definitions into each block if lazy is in kwargs and is False;
         otherwise, do not load the definitions - they'll be loaded later when needed.
         """
+        lazy = kwargs.pop('lazy', True)
+        should_cache_items = not lazy
+
         runtime = self._get_cache(course_entry.structure['_id'])
         if runtime is None:
-            lazy = kwargs.pop('lazy', True)
             runtime = self.create_runtime(course_entry, lazy)
             self._add_cache(course_entry.structure['_id'], runtime)
+            should_cache_items = True
+
+        if should_cache_items:
```

Comment:
**Note to reviewers:** This change allows callers to be agnostic to whether a course was previously prefetched into the runtime.  This is No 2 in the description.